lib/ts/recipe/multifactorauth/types.ts

```diff
@@ -0,0 +1,136 @@
+};
+
+// Config is what does in the constructor of the recipe.
+export type UserInput = {
+    firstFactors?: string[];
+    getSecondaryFactorInfo?: (
+        builtInFactors: SecondaryFactorRedirectionInfo[],
+        userContext: UserContext
+    ) => SecondaryFactorRedirectionInfo[];
+    disableDefaultUI?: boolean;
+    factorChooserScreen?: FeatureBaseConfig;
+
+    override?: {
+        functions?: (
+            originalImplementation: RecipeInterface,
+            builder?: OverrideableBuilder<RecipeInterface>
+        ) => RecipeInterface;
+    };
+} & RecipeModuleUserInput<GetRedirectionURLContext, PreAndPostAPIHookAction, OnHandleEventContext>;
+
+// Config is what does in the constructor of the recipe.
+export type Config = UserInput &
+    RecipeModuleConfig<GetRedirectionURLContext, PreAndPostAPIHookAction, OnHandleEventContext>;
+
+export type NormalisedConfig = {
+    firstFactors?: string[];
+    getSecondaryFactorInfo: (
+        builtInFactors: SecondaryFactorRedirectionInfo[],
+        userContext: UserContext
+    ) => SecondaryFactorRedirectionInfo[];
+    disableDefaultUI: boolean;
+    factorChooserScreen: FeatureBaseConfig;
+
+    override: {
+        functions: (
+            originalImplementation: RecipeInterface,
+            builder?: OverrideableBuilder<RecipeInterface>
+        ) => RecipeInterface;
+    };
+} & NormalisedRecipeModuleConfig<GetRedirectionURLContext, PreAndPostAPIHookAction, OnHandleEventContext>;
+
+export type GetRedirectionURLContext =
+    | {
+          action: "FACTOR_CHOOSER";
+          nextFactorOptions?: string[];
+      }
+    | {
+          action: "GO_TO_FACTOR";
+          factorId: string;
+          forceSetup?: boolean;
+      };
+
+export type PreAndPostAPIHookAction = "GET_MFA_INFO";
+
+export type PreAPIHookContext = {
+    action: PreAndPostAPIHookAction;
+    requestInit: RequestInit;
+    url: string;
+    userContext: UserContext;
+};
+
+export type OnHandleEventContext = {
+    action: "FACTOR_CHOOSEN";
+    factorId: string;
+};
+
+export type LoadedMFAInfo = {
+    factors: MFAFactorInfo;
+    emails: Record<string, string[] | undefined>;
+    phoneNumbers: Record<string, string[] | undefined>;
+};
+
+export type FactorChooserThemeProps = {
+    mfaInfo: LoadedMFAInfo;
+    availableFactors: SecondaryFactorRedirectionInfo[];
+    showBackButton: boolean;
+    onBackButtonClicked: () => void;
+    navigateToFactor: (factorId: string) => void;
+    onLogoutClicked: () => void;
+    config: NormalisedConfig;
+    userContext?: UserContext;
+};
+
+export type SecondaryFactorRedirectionInfo = {
+    id: string;
+    name: string;
+    description: string;
+    logo: FC;
+    path: string;
+};
+
+export const FactorIds = {
+    EMAILPASSWORD: "emailpassword",
+    OTP_EMAIL: "otp-email",
+    OTP_PHONE: "otp-phone",
+    LINK_EMAIL: "link-email",
+    LINK_PHONE: "link-phone",
+    THIRDPARTY: "thirdparty",
+    TOTP: "totp",
+};
```

Comment:
need to write in pr checklist that if we add a new factor, this needs to be modified.